src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -8,56 +13,96 @@ import { HasuraService as HasuraServiceFromServices } from '../services/hasura/h
 import { KeycloakService } from '../services/keycloak/keycloak.service';
 @Injectable()
 export class BeneficiariesService {
-    public url = process.env.HASURA_BASE_URL;
-
- 
-    constructor(
-       
-      private readonly httpService: HttpService,
-      private userService:UserService,
-      private helper: UserHelperService,
-      private hasuraService: HasuraService,
-      private hasuraServiceFromServices: HasuraServiceFromServices,
-      private keycloakService: KeycloakService,
-      private configService: ConfigService
-    ){}
-
-    public returnFields=[
-        "status",
-        "user_id",
-        "enrollment_number",
-        "beneficiaries_found_at",
-        "documents_status",
-        "enrollment_status",
-        "enrolled_for_board",
-        "type_of_enrollement",
-        "academic_year_id",
-        "payment_receipt_document_id",
-        "facilitator_id",
-        "documents_status",
-        "program_id",
-        "reason_for_status_update",   
-        "created_by",
-        "updated_by",
-        
-    ]
-      
-    public async findAll(body: any,req:any,resp:any) {
-        const user=await this.userService.ipUserInfo(req)
-            const { status,sortType } = body;
-            const page = body.page ? body.page : '1';
-            const limit = body?.limit ? body?.limit : '10';
-        
-            let offset = 0;
-            if (page > 1 && limit) {
-              offset = parseInt(limit) * (page - 1);
-            }
-            let query =''
-        if(status){
-          let query = `{beneficiaries:{status:{_eq:${status}}}}`;
+  public url = process.env.HASURA_BASE_URL;
+
+  constructor(
+    private readonly httpService: HttpService,
+    private userService: UserService,
+    private helper: UserHelperService,
+    private hasuraService: HasuraService,
+    private hasuraServiceFromServices: HasuraServiceFromServices,
+    private keycloakService: KeycloakService,
+    private configService: ConfigService,
+  ) {}
+
+  public returnFields = [
+    'status',
+    'user_id',
+    'enrollment_number',
+    'beneficiaries_found_at',
+    'documents_status',
+    'enrollment_status',
+    'enrolled_for_board',
+    'type_of_enrollement',
+    'academic_year_id',
+    'payment_receipt_document_id',
+    'facilitator_id',
+    'documents_status',
+    'program_id',
+    'reason_for_status_update',
+    'created_by',
+    'updated_by',
+  ];
+
+  //status count
+  public async getStatuswiseCount(req: any, resp: any) {
+    const user = await this.userService.ipUserInfo(req);
+    const status = [
+      'identified',
+      'ready_to_enroll',
+      'enrolled',
+      'approved_ip',
+      'registered_in_camp',
+      'pragati_syc',
+      'rejected',
+      'dropout',
+    ];
+    let qury = `query MyQuery {
+      ${status.map(
+        (
+          item,
+        ) => `${item}:beneficiaries_aggregate(where: {status: {_eq: "${item}"}}) {
```

Comment:
add where condition where beneficiary_id = logged in prerak's id

Reply:
resolved sir